Implement a slightly better way to find libswiftReflection.a

adrian-prantl opened the pull request.

dcci commented:
We really don't need the bits in `source/Symbol/SwiftASTContext.cpp`, I'll remove them, sorry.
BTW, the patch itself looks good

dcci commented:
(except that maybe it needs to be rebased)

adrian-prantl committed:
Linux / cmake: Install the clang resource directory to the expected POSIX path.

HostInfoPosix::ComputeClangDirectory() expects the clang resource
directory to be in lib/lldb/clang/$VERSION. This patch fixes the
Swift-specific CMake to install it there instead of in lib/lldb/clang/.

<rdar://problem/37354542>
https://bugs.swift.org/browse/SR-6954

adrian-prantl committed:
Linux / cmake: Install the clang resource directory to the expected POSIX path.

HostInfoPosix::ComputeClangDirectory() expects the clang resource
directory to be in lib/lldb/clang/$VERSION. This patch fixes the
Swift-specific CMake to install it there instead of in lib/lldb/clang/.

<rdar://problem/37354542>
https://bugs.swift.org/browse/SR-6954

(cherry picked from commit 0f090945431a91eb0a6a89a7c8dd85b0c0d5c1f8)

Commit:
[SwiftLanguageRuntime] Ask the Demangler, don't reinvent the wheel.

It's really not lldb business that of implementing the logic to
understand whether a mangled name is that of a swift class. This
removes a bunch of duplicated code, making the codepath slightly
more robust as we now ask the demangler.

<rdar://problem/37710513>

donhinton committed:
[cmake] Fix LLDB_CODESIGN_IDENTITY logic.

Summary:
Consolidate LLDB_CODESIGN_IDENTITY logic in one place and use
SKIP_DEBUGSERVER, which can be set independently, to control
codesigning targets.

Currently, running cmake the first time in a clean directory, without
passing -DLLDB_CODESIGN_IDENTITY='', fails.  However, subsequent runs
succeed.  That's because LLDB_CODESIGN_IDENTITY gets added to the
CACHE after the initial test.  To fix that, the default value must be
set before it's tested.

Here's the error produced on the first run:

CMake Error at tools/lldb/tools/debugserver/source/CMakeLists.txt:215 (add_custom_command):
  No TARGET 'debugserver' has been created in this directory.

Differential Revision: https://reviews.llvm.org/D43432

git-svn-id: https://llvm.org/svn/llvm-project/lldb/trunk@325442 91177308-0d34-0410-b5e6-96231b3b80d8
(cherry picked from commit 6329f267e05765db1c1c6ec50d350abd9f70f077)

 Conflicts:
	tools/debugserver/source/CMakeLists.txt

donhinton committed:
Add DEBUGSERVER_PATH to the cache so it'll be available for tests.

This fixed a problem caused by r325442.

Differential Revision: https://reviews.llvm.org/D43432

git-svn-id: https://llvm.org/svn/llvm-project/lldb/trunk@325452 91177308-0d34-0410-b5e6-96231b3b80d8
(cherry picked from commit 85a036ba61e7efe66caac2ed0cf33e6ff05f3572)

JDevlieghere committed:
Handle typeof() expressions

Before this patch, LLDB was not able to evaluate expressions that
resulted in a value with a typeof- or decltype-type. This patch fixes
that.

Before:
  (lldb) p int i; __typeof__(i) j = 1; j
  (typeof (i)) $0 =

After:
  (lldb) p int i; __typeof__(i) j = 1; j
  (typeof (i)) $0 = 1

Differential revision: https://reviews.llvm.org/D43471

rdar://37461520

git-svn-id: https://llvm.org/svn/llvm-project/lldb/trunk@325568 91177308-0d34-0410-b5e6-96231b3b80d8
(cherry picked from commit 98516a2168f6d61fee2c1e53ec48b26674ba9659)

vedantk committed:
[ExpressionParser] Clean up SwiftREPL::CreateInstance

This is a mostly-NFC change which makes SwiftREPL::CreateInstance easier
to read with early returns and references.

The only functional change is a bugfix for the call to
GetScratchTypeSystemForLanguage. If that call fails, we will now
correctly tear down the REPL process and the event handler thread.

Commit:
[SwiftRepl] Improve the CMake build & simplify.

This is a grab bag of improvements for the repl. There are
several issues.
1) The repl needs a dummy target to run. There were two, one
written in C, the other in swift. The latter doesn't really
work and the only reason why nobody noticed is that only the
CMake build used it and there was no test whatsoever. The good
news is that's unneeded and can be removed.
2) The way the dummy executable was built with CMake was wrong
and unnecessarily complicated. This commit rewrites the logic
to be more readable (and commented).
3) This logic expects optimizations to be turned off, but
that was delegated to the build system. Now it's explicit
in the code via attributes, so that we don't have to duplicate
the logic across CMake/XCodebuild.

The logic is still fairly fragile as we dlopen() a library
looking inside @rpath. After speaking with Louis, he told me
that we shouldn't rely on this behaviour (and I tend to agree).
It's more complicated to fix because it involves touching also
the xcodebuild. This is left as a follow-up.

jimingham committed:
This test is flakey on Linux, disable it.

The code has varied significantly on TOT so fixing
it on stable is not worth the effort.

<rdar://problem/37725893>

jimingham committed:
Merge pull request #336 from jimingham/disable-unittest

This test is flakey on Linux, disable it.

dcci committed:
Merge pull request #335 from dcci/repl-test

[SwiftRepl] Improve the CMake build & simplify.

dcci committed:
Merge pull request #330 from dcci/patatino

[SwiftLanguageRuntime] Ask the Demangler, don't reinvent the wheel.

JDevlieghere committed:
Merge pull request #339 from JDevlieghere/stable

Handle typeof() expressions

adrian-prantl committed:
Merge pull request #331 from adrian-prantl/37354542

 Linux / cmake: Install the clang resource directory to the expected POSIX path

Commit:
[REPL] Rewrite dictionary testing & improve infra to be more reliable.

This now uses lit. It's shorter and hopefully more reliable than
pexpect. As part of this, I also did all the plumbing for testing
the repl in lldb, so what I've done here can be used for other repl
tests. While I'm at it, remove the old test, we don't need it anymore.

<rdar://problem/37407231>

keith committed:
Fix python build argument

Previously this concatenated "pythonswift/utils" this was broken in
80e032cb3fc68d1c56ed0c112f1fc81baf1a9454

adrian-prantl committed:
Merge pull request #337 from adrian-prantl/37354542-4.1

Linux / cmake: Install the clang resource directory to the expected P…

dcci committed:
Merge pull request #340 from dcci/litdict

[REPL] Rewrite dictionary testing & improve infra to be more reliable.

Commit:
[REPL] Replace a flakey NSString test with a robust version.

This was disabled, sadly because it was flakey.
The new version is shorter and more reliable.

<rdar://problem/37407231>

adrian-prantl committed:
Merge branch 'swift-4.1-branch' of github.com:apple/swift-lldb into stable

dcci committed:
Merge pull request #343 from dcci/nsstring_repl

[REPL] Replace a flakey NSString test with a robust version.

dduan committed:
Merge pull request #341 from keith/ks/fix-build

Fix python build argument

vedantk committed:
[ObjC] Fix the NSConcreteData formatter and test it

The length field of an NSConcreteData lives one word past the start of
the object, not two.

rdar://37670464

git-svn-id: https://llvm.org/svn/llvm-project/lldb/trunk@325841 91177308-0d34-0410-b5e6-96231b3b80d8
(cherry picked from commit ed9fb0ceca0d4e08ab4372eb4cfa9bbbc215272a)

adrian-prantl committed:
Merge pull request #344 from adrian-prantl/merge

Merge conflict resolution.

vedantk committed:
Delete some unused #includes of CleanUp.h, NFC

git-svn-id: https://llvm.org/svn/llvm-project/lldb/trunk@325847 91177308-0d34-0410-b5e6-96231b3b80d8
(cherry picked from commit f4b912736fbe899263038aacaefae24fb178d807)

adrian-prantl committed:
Revert "Linux / cmake: Install the clang resource directory to the expected POSIX path."

This reverts commit c29f16288bc2fd3959cd1eef13fc6c41b6051800
because it caused

https://bugs.swift.org/browse/SR-7065
REPL broken on Linux (cannot import anything)

vedantk committed:
[Utility] Simplify and generalize the CleanUp helper, NFC

Removing the template arguments and most of the mutating methods from
CleanUp makes it easier to understand and reuse.

In its present state, CleanUp would be too cumbersome to adapt to cases
where multiple objects need to be released. Take for example this change
in swift-lldb:

  https://github.com/apple/swift-lldb/pull/334/files#diff-6f474df750f75c8ba675f2a8408a5629R219

This change is simple to express with the new CleanUp, but not so simple
with the old version.

Differential Revision: https://reviews.llvm.org/D43662

git-svn-id: https://llvm.org/svn/llvm-project/lldb/trunk@325964 91177308-0d34-0410-b5e6-96231b3b80d8
(cherry picked from commit b438a3f143199fa44630195d103d465e05425c3b)

 Conflicts:
	lldb.xcodeproj/project.pbxproj
	source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp
	unittests/Utility/CMakeLists.txt

vedantk committed:
Use the new CleanUp utility, NFC

adrian-prantl committed:
Merge pull request #346 from adrian-prantl/37354542-4.1-revert

Revert "Linux / cmake: Install the clang resource directory to the ex…

adrian-prantl committed:
Merge branch 'swift-4.1-branch' of github.com:apple/swift-lldb into stable

adrian-prantl committed:
Merge pull request #348 from adrian-prantl/revert

Merge conflict resolution.

vedantk committed:
[REPL] Replace the flaky NSObjectSubclass test

This test did not reliably complete on the bots and has strange baked-in
timeouts. Convert it to a lit test, which should be more reliable.

rdar://36628021

adrian-prantl committed:
Set DebuggerSupport to true in the ClangImporterOptions.

Swift's ClangImporter is more lenient when this option is set, but
LLDB wasn't passing it correctly to the ClangImporter, only to the
LanguageOptions.

<rdar://problem/37922760>

vedantk committed:
[REPL] test: Convert TestREPLRecursiveClass.py to a lit test

vedantk committed:
[REPL] test: Convert TestREPLTypeLookup.py to a lit test

vedantk committed:
[REPL] test: Convert TestREPLStructs.py to a lit test

vedantk committed:
[REPL] test: Convert TestREPLQuickLookObject.py to a lit test

vedantk committed:
[REPL] test: Convert TestREPLClasses.py to a lit test

vedantk committed:
[REPL] test: Add a lit test for REPL breakpoints

vedantk committed:
[REPL] Consolidate 6 tests into one lit test

vedantk committed:
[REPL] test: Convert TestPOUnwrapping.py to a lit test

vedantk committed:
[REPL] test: Clean up the test layout

- Split some of the testing in RecursiveClass.test out into Class.test.
- Rename the SwiftExpr test directory to SwiftREPL.
- Rename TypeLookup.test to ImportFoundation.test.

vedantk committed:
[REPL] test: Convert TestREPLBasic.py to a lit test

adrian-prantl committed:
Merge pull request #351 from adrian-prantl/37922760

 Set DebuggerSupport to true in the ClangImporterOptions.

adrian-prantl committed:
Convert if to early exit (NFC)

adrian-prantl committed:
Refactor SwiftASTContext::CreateInstance into two separate functions
for creating a SwiftASTContext from either a Target or a Module.

vedantk committed:
[REPL] Build the dummy repl on Linux for testing

adrian-prantl committed:
Merge pull request #353 from adrian-prantl/cleanup

Cleanup

vedantk committed:
[test] Delete TestSwiftFoundation.py, which was accidentally left behind

This test has no accompanying source code, and is incomplete. It must
have been accidentally left behind.

vedantk committed:
[test] Treat inline tests as part of the swiftpr category

1) Treat all inline tests as if they are a part of the swiftpr category.
This means "dotest.py -G swiftpr" will include all inline tests, unless
they are disabled in the MakeInlineTest constructor.

2) In the MakeInlineTest constructor, only enable a test if it has an
appropriate debug info kind. E.g, don't enable the dsym variant of a
test if "--skip-categories dsym" has been specified.

3) In the MakeInlineTest constructor, treat all tests as a part of the
swiftpr category *unless* they have decorators. An inline test with
decorators is usually either skipped on some platform or x-failed: for
now, we will not consider these tests for PR testing.

4) Remove the 'swiftpr' decorators from inline tests as they are not
reliable. They cause the wrong set of tests to be included in the PR
testing category.

huonw committed:
[ExpressionParser] swiftc change: OwnershipAttr -> ReferenceOwnershipAttr.

Commit:
[Types] Initial plumbing for using remote mirrors in the debugger.

<rdar://problem/36477650>

[RemoteMirrors] Switch readBytes to conform to the new API.

Commit:
[Repl] Don't crash when passing additional arguments.

The assertion was overly aggressive, we can actually
have a non-empty set of options in this codepath. While
here, add a test to make sure this doesn't regress.

<rdar://38042866>

dcci committed:
Merge pull request #356 from dcci/replcrash

[Repl] Don't crash when passing additional arguments.

shahmishal committed:
Merge pull request #355 from huonw/ownership-rename

[ExpressionParser] swiftc change: OwnershipAttr -> ReferenceOwnershipAttr.

Commit:
[REPL] Build the dummy executable always with `-g0`.

This is just an implementation detail, and nobody touches it
anyway. Given that's a C program, lldb frame language recognition
gets confused, so the repl malfunctions. It only manifests in
debug builds, that's the reason why nobody noticed (until now).

An alternative would be that of having the dummy executable
being written in swift. At some point this was the case, but
then it was converted to a C program. Eventually we might want
to re-evaluate the choice.

<rdar://problem/38121735>

dcci committed:
Merge pull request #360 from dcci/rdar38121735

[REPL] Build the dummy executable always with `-g0`.

adrian-prantl committed:
Merge branch 'remotemirrorsnew' of https://github.com/dcci/swift-lldb into stable

adrian-prantl committed:
WIP: Add reflection images for all dylibs
